src/cache.cc

```diff
@@ -2,6 +2,11 @@
 #include "set.h"
 
 uint64_t l2pf_access = 0;
+/* you can implement each page walk cache level as a 2d array */
+uint64_t pml4[PML4_SET][PML4_WAY], pdp[PDP_SET][PDP_WAY], pd[PD_SET][PD_WAY];
+
+/* implementation of LRU policy with counters */
+uint64_t pml4_lru[PML4_SET][PML4_WAY], pdp_lru[PDP_SET][PDP_WAY], pd_lru[PD_SET][PD_WAY];
```

Comment:
My preferred style is to create structs for each table entry to keep the data grouped together.
```suggestion
/* you can implement each page walk cache level as a 2d array */
struct pt_entry
{
    uint64_t address;
    uint64_t lru;
};
pt_entry pml4[PML4_SET][PML4_WAY], pdp[PDP_SET][PDP_WAY], pd[PD_SET][PD_WAY];
```

This makes debugging easier (data is logically grouped with its metadata) and in my opinion more accurately represents the way we think about this data.

Reply:
I agree. Much cleaner in my view.